tests/src/App/IndexedName.cpp

```diff
@@ -0,0 +1,243 @@
+}
+
+TEST_F(IndexedNameTest, preincrementOperator)
+{
+    auto indexedName = Data::IndexedName("EDGE42");
+    ++indexedName;
+    EXPECT_EQ(indexedName.getIndex(), 43);
+}
+
+TEST_F(IndexedNameTest, predecrementOperator)
+{
+    auto indexedName = Data::IndexedName("EDGE42");
+    --indexedName;
+    EXPECT_EQ(indexedName.getIndex(), 41);
+}
+
+TEST_F(IndexedNameTest, comparisonOperators)
+{
+    auto indexedName1 = Data::IndexedName("EDGE42");
+    auto indexedName2 = Data::IndexedName("EDGE42");
+    EXPECT_EQ(indexedName1.compare(indexedName2), 0);
+    EXPECT_TRUE(indexedName1 == indexedName2);
+    EXPECT_FALSE(indexedName1 != indexedName2);
+    EXPECT_FALSE(indexedName1 < indexedName2);
+
+    auto indexedName3 = Data::IndexedName("EDGE42");
+    auto indexedName4 = Data::IndexedName("FACE42");
+    EXPECT_EQ(indexedName3.compare(indexedName4), -1);
+    EXPECT_FALSE(indexedName3 == indexedName4);
+    EXPECT_TRUE(indexedName3 != indexedName4);
+    EXPECT_TRUE(indexedName3 < indexedName4);
+
+    auto indexedName5 = Data::IndexedName("FACE42");
+    auto indexedName6 = Data::IndexedName("EDGE42");
+    EXPECT_EQ(indexedName5.compare(indexedName6), 1);
+    EXPECT_FALSE(indexedName5 == indexedName6);
+    EXPECT_TRUE(indexedName5 != indexedName6);
+    EXPECT_FALSE(indexedName5 < indexedName6);
+
+    auto indexedName7 = Data::IndexedName("EDGE41");
+    auto indexedName8 = Data::IndexedName("EDGE42");
+    EXPECT_EQ(indexedName7.compare(indexedName8), -1);
+    EXPECT_FALSE(indexedName7 == indexedName8);
+    EXPECT_TRUE(indexedName7 != indexedName8);
+    EXPECT_TRUE(indexedName7 < indexedName8);
+
+    auto indexedName9 = Data::IndexedName("EDGE43");
+    auto indexedName10 = Data::IndexedName("EDGE42");
+    EXPECT_EQ(indexedName9.compare(indexedName10), 1);
+    EXPECT_FALSE(indexedName9 == indexedName10);
+    EXPECT_TRUE(indexedName9 != indexedName10);
+    EXPECT_FALSE(indexedName9 < indexedName10);
+}
+
+TEST_F(IndexedNameTest, subscriptOperator)
+{
+    // operator[]
+    auto indexedName = Data::IndexedName("EDGE42");
+    EXPECT_EQ(indexedName[0], 'E');
+    EXPECT_EQ(indexedName[1], 'D');
+    EXPECT_EQ(indexedName[2], 'G');
+    EXPECT_EQ(indexedName[3], 'E');
+}
+
+TEST_F(IndexedNameTest, getType)
+{
+    auto indexedName = Data::IndexedName("EDGE42");
+    EXPECT_STREQ(indexedName.getType(), "EDGE");
+}
+
+TEST_F(IndexedNameTest, setIndex)
+{
+    auto indexedName = Data::IndexedName("EDGE42");
+    EXPECT_EQ(indexedName.getIndex(), 42);
+    indexedName.setIndex(1);
+    EXPECT_EQ(indexedName.getIndex(), 1);
+}
+
+TEST_F(IndexedNameTest, isNull)
+{
+    auto indexedName = Data::IndexedName();
+    EXPECT_TRUE(indexedName.isNull());
+    indexedName = Data::IndexedName("",1);
+    EXPECT_TRUE(indexedName.isNull());
+    indexedName = Data::IndexedName("EDGE",1);
+    EXPECT_FALSE(indexedName.isNull());
+    indexedName = Data::IndexedName("INVALID42NAME",1);
+    EXPECT_TRUE(indexedName.isNull());
```

Comment:
IMO, what's missing here is a test with a string with non-letter characters, e.g. `indexedName = Data::IndexedName(".EDGE",1);`

Reply:
Good idea, thanks.

Reply:
Added in d4d9ce7c18176d113713c330882ca4a10b68d55a